tflint/runner.go

```diff
@@ -501,6 +502,69 @@ func (r *Runner) LookupIssues(files ...string) Issues {
 	return issues
 }
 
+// WalkExpressions visits all blocks that can contain expressions:
+// resource, data, module, provider, locals, and output. It calls the walker
+// function with every expression it encounters and halts if the walker
+// returns an error.
+func (r *Runner) WalkExpressions(walker func(hcl.Expression) error) error {
```

Comment:
👍 

It would be better if it has tests.

Reply:
Will do!